src/main/java/com/shapesecurity/shift/es2016/semantics/Explicator.java

```diff
@@ -463,42 +466,54 @@ Node explicateStatement(@Nonnull Statement statement, boolean strict) {
 			targets = targets.put(statement, new Pair<>(outerTarget, Maybe.of(innerTarget)));
 			Break breakNode = new Break(outerTarget, ImmutableList.empty());
 
-			return makeUnvaluedTemporary(object -> makeUnvaluedTemporary(counter -> makeUnvaluedTemporary(keys -> {
-				Block body = new Block(ImmutableList.<Node>of(
-					new IfElse(
-						new RelationalComparison(
-							RelationalComparison.Operator.LessThan,
-							counter,
-							new MemberAccess(keys, new LiteralString("length"))
-						),
-						new Block(Void.INSTANCE),
-						new Block(breakNode)
-					))
-					.append(ImmutableList.of(
-						new IfElse(
-							new In(new MemberAccess(keys, counter), object),
+			return makeUnvaluedTemporary(
+				object -> makeUnvaluedTemporary(
```

Comment:
Can you move the newline to following the `->`?

Reply:
Technically yes, but that would be inconsistent with the rest of this file, so I'd prefer not to.